### PR TITLE
Renamed RESTBase to HyperSwitch everywhere

### PR DESCRIPTION
**Pull request by Pchelolo**
Just renaming `RESTBase` to `HyperSwitch` everywhere it's needed.

### CODE REVIEW

**Comment by d00rman** [comment]
Ok, let's bike-shed a bit: Hyperswitch or HyperSwitch? My :+1: goes to the former, CamelCase is so 2000's :)

**Comment by Pchelolo** [comment]
@d00rman May be `HyPeRsWiTcH`? :) 
I'm OK with either Hyperswitch and HyperSwitch, @gwicke @eevans what's your preference?

**Comment by d00rman** [comment]
> May be HyPeRsWiTcH? :)

Yeah! :100:

**Comment by gwicke** [comment]
To me HyperSwitch feels most natural in a context where Capitalization is expected.

File: lib/server.js [hunk 1]
```diff
@@ -254,7 +254,7 @@ function handleResponse(opts, req, resp, response) {
         resp.writeHead(response.status || 500, '', response.headers);
         resp.end(req.method === 'head' ? undefined : JSON.stringify({
             type: 'https://restbase.org/errors/no_content',
-            title: 'RESTBase error: No content returned by backend.',
+            title: 'Hyperswitch error: No content returned by backend.',
```
Comment (gwicke) [inline lib/server.js]: Switch ;)

File: lib/auth.js [hunk 2]
```diff
@@ -23,20 +23,20 @@ function constructInternalRequestRegex(reqWhiteList) {
 }
 
 /**
- * Copies forwarded headers from restbase to request.
+ * Copies forwarded headers from Hyperswitch to request.
```
Comment (gwicke) [inline lib/auth.js]: HyperSwitch

File: lib/hyperswitch.js [hunk 3]
```diff
@@ -362,8 +362,8 @@ RESTBase.prototype._request = function(req, options) {
     }
 
     if (handler) {
-        // Prepare to call the handler with a child restbase instance
-        var childRESTBase = this.makeChild(childReq, options);
+        // Prepare to call the handler with a child HyperSwitch instance
+        var childHyper = this.makeChild(childReq, options);
```
Comment (gwicke) [inline lib/hyperswitch.js]: Minor: Could consider using childHyperSwitch here.

File: lib/router.js [hunk 4]
```diff
@@ -207,18 +207,18 @@ Router.prototype._loadModule = function(modDef, globals) {
     }
 };
 
-Router.prototype._loadModules = function(node, restBaseModules, scope) {
+Router.prototype._loadModules = function(node, hsModules, scope) {
```
Comment (gwicke) [inline lib/router.js]: hyperModules?

File: test/framework/handlerTemplate/test_config.yaml [hunk 5]
```diff
@@ -82,7 +82,7 @@ spec_root: &spec_root
 
 # Finally, a standard service-runner config.
 info:
-  name: restbase
+  name: hyperswitch
```
Comment (d00rman) [inline test/framework/handlerTemplate/test_config.yaml]: This should be left as-is.

File: test/framework/handlerTemplate/test_config.yaml [hunk 6]
```diff
@@ -92,5 +92,5 @@ services:
       spec: *spec_root
       salt: secret
       default_page_size: 1
-      user_agent: RESTBase-testsuite
+      user_agent: HyperSwitch-testsuite
```
Comment (d00rman) [inline test/framework/handlerTemplate/test_config.yaml]: Idem.

File: test/framework/handlerTemplate/test_config.yaml (hunk 5 above)
Comment (d00rman) [inline test/framework/handlerTemplate/test_config.yaml]: Actually, nevermind. It's ok like this.

Comment (Pchelolo) [inline test/framework/handlerTemplate/test_config.yaml]: @d00rman why? This is a framework test, it has nothing to do with RESTBase

File: lib/router.js (hunk 4 above)
Comment (d00rman) [inline lib/router.js]: +1

File: lib/hyperswitch.js (hunk 3 above)
Comment (d00rman) [inline lib/hyperswitch.js]: euh, `childHyperSwitch` is a bit too long, and `childHyper` is more in line with the var name `hyper`, IMHO.

Comment (d00rman) [inline lib/hyperswitch.js]: I was too late. @Pchelolo already renamed it :/

**Comment by gwicke** [comment]
LGTM. If Travis agrees, I'd say lets get this merged.

**Pull request merged**